File: Domain/Entities/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CarRentail.Domain.Interface;

namespace CarRentail.Domain.Entities
{
    public class Vehicle 
    {
        public int Id { get; set; }
        public string Brand { get; set; }
        public string CarNumber { get; set; }
        public string Model { get; set; }
        public int Year { get; set; }
        public int Distance { get; set; }
        public string Photo { get; set; }
        public int Price { get; set; }
        public int EnginePower { get; set; }
        public int ElectricPower { get; set; }
        public int BatteryCapacity { get; set; }
        public bool State { get; set; }
        //public string VehicleType { get; set; }
    }
}

File: CarRentGUI/src/app/Models/Vehicle.ts
export class Vehicle {
    id: number;
    brand: string;
    carNumber: string;
    model: string;
    year: number;
    distance: number;
    photo: string;
    price: number;
    enginePower: number;
    electricPower!: number;
    batteryCapacity!: number;
    vehicleType: string;

    constructor(
        id: number,
        brand: string,
        carNumber: string,
        model: string,
        year: number,
        distance: number,
        photo: string,
        price: number,
        enginePower: number,
        electricPower: number,
        batteryCapacity: number,
        vehicleType: string
    ) {
        this.id = id;
        this.brand = brand;
        this.carNumber = carNumber;
        this.model = model;
        this.year = year;
        this.distance = distance;
        this.photo = photo;
        this.price = price;
        this.enginePower = enginePower;
        this.electricPower = electricPower;
        this.batteryCapacity = batteryCapacity;
        this.vehicleType = vehicleType;
    }
}
